src/DB.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Handles the database.
+ *
+ * @package juvo\AS_Processor
+ */
+
+namespace juvo\AS_Processor;
+
+use juvo\AS_Processor\Entities\ProcessStatus;
+
+/**
+ * Database methods.
+ */
+trait DB
+{
+    /**
+     * Chunks table name
+     *
+     * @var string
+     */
+    public string $table_name_chunks = 'asp_chunks';
+
+    /**
+     * Track if table has been checked
+     *
+     * @var boolean
+     */
+    private bool $table_checked = false;
+
+    /**
+     * Gets the table name for the chunks.
+     *
+     * @return string
+     */
+    public function get_chunks_table_name(): string
+    {
+        return $this->db()->prefix . $this->table_name_chunks;
+    }
+
+    /**
+     * Gets the wpdb wrapper.
+     *
+     * @return \wpdb the WordPress database wrapper.
+     */
+    public function db(): \wpdb
+    {
+        global $wpdb;
+        if (!$this->table_checked) {
+            $this->maybe_create_table($wpdb);
+            $this->table_checked = true;
+        }
+        return $wpdb;
+    }
+
+    /**
+     * Check if table exists and create if not.
+     *
+     * @param \wpdb $wpdb the WordPress database wrapper.
+     * @return void
+     */
+    private function maybe_create_table( \wpdb $wpdb ): void
+    {
+        $table_name = $wpdb->prefix . $this->table_name_chunks;
+
+        // Get all enum values and make them sql friendly by wrapping them in quotes
+        $enum_values = array_map(
+            static fn(ProcessStatus $status): string => sprintf("'%s'", $status->value),
+            ProcessStatus::cases()
+        );
+        $enum_definition = implode(',', $enum_values);
+
+        $charset_collate = $wpdb->get_charset_collate();
+        $sql = "CREATE TABLE {$table_name} (
+            `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+            `action_id` bigint(20) unsigned DEFAULT NULL,
+            `name` varchar(255) NOT NULL,
+            `group` varchar(255) NOT NULL,
+            `status` ENUM({$enum_definition}) NOT NULL,
+            `data` longtext NOT NULL,
+            `start` decimal(14,4) DEFAULT NULL,
+            `end` decimal(14,4) DEFAULT NULL,
+            PRIMARY KEY (`id`),
+            KEY `status` (`status`),
+            KEY `start` (`start`),
+            KEY `end` (`end`),
+            KEY `action_id` (`action_id`)
+        ) {$charset_collate}";
```

Comment:
i like what you tried and appreciate the effort, but this will likely cause errors once the enum options change. I think in this particular case storing the string and also parsing it back is the better option.

Ideally in a perfect world we would have a database migration that we could use to add new enum options as well

Reply:
Also i just found that SQlite does not support enums